sender/http_sender_test.go

```diff
@@ -11,48 +11,132 @@ import (
 )
 
 func TestSendSimplePostMessage(t *testing.T) {
+	// Given
 	req := "{\"name\": \"test\"}"
 
-	// Send response to be tested
 	server := httptest.NewServer(createHandler(http.StatusOK, req))
+	request := []config.Request{createRequest("test", req, 1, server)}
 	defer server.Close()
 
-	sender := prepareSender(req, server)
+	// When
+	sender := prepareSender(request, server)
+	result, err := sender.SendWithResponse()
 
-	err := sender.Send()
+	// Then
 	if err != nil {
 		t.Fatalf("expected a no error, instead got: %s", err.Error())
 	}
+	for r := range result {
+		if r.Metadata["status"] != "200 OK" {
+			t.Fatalf("expected a 200 status, instead got: %s", r.Metadata["status"])
+		}
+	}
 }
 
-func TestRecievedServerError(t *testing.T) {
+func TestSimpleGetMessageWhenResponseIs500(t *testing.T) {
+	// Given
 	req := "{\"name\": \"test\"}"
 
-	// Send response to be tested
 	server := httptest.NewServer(createHandler(http.StatusInternalServerError, req))
+	request := []config.Request{createRequest("test", req, 1, server)}
+	defer server.Close()
+
+	// When
+	sender := prepareSender(request, server)
+	result, err := sender.SendWithResponse()
+	count := 0
+
+	// Then
+	if err != nil {
+		t.Fatalf("expected a no error, instead got: %s", err.Error())
+	}
+	for r := range result {
+		count++
+		if r.Metadata["status"] != "500 Internal Server Error" {
```

Comment:
BTW, it is possible to make an alias for map, call it something like "ResponseMetadata", then use it as receiver in functions, so that it can be
r.Metdata.Status() == 500